src/util/Injected/Utils.js

```diff
@@ -587,10 +587,13 @@ exports.LoadUtils = () => {
             model.isGroup = true;
             const chatWid = window.Store.WidFactory.createWid(chat.id._serialized);
             await window.Store.GroupMetadata.update(chatWid);
-            chat.groupMetadata.participants._models
-                .filter(x => x.id?._serialized?.endsWith('@lid'))
-                .forEach(x => x.contact?.phoneNumber && (x.id = x.contact.phoneNumber));
             model.groupMetadata = chat.groupMetadata.serialize();
+            model.groupMetadata.participants = chat.groupMetadata.participants._models.map(item => {
+                const result = item.serialize();
+                result.lid = result.id;
+                result.id = item.contact?.phoneNumber || result.lid;
```

Comment:
I suggest adding a check so that lid is only set when the result.id is actually of type lid.
This avoids adding unnecessary properties for @c.us or @g.us Ids:

```suggestion
                if (result.id.server === 'lid') {
                    result.lid = result.id;
                    result.id = item.contact?.phoneNumber || result.lid;
                }              
```